acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -494,11 +471,15 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> ACVM<'a, F, B> {
     /// 1. All opcodes have been executed successfully.
     /// 2. The circuit has been found to be unsatisfiable.
     /// 2. A Brillig [foreign call][`ForeignCallWaitInfo`] has been encountered and must be resolved.
-    pub fn solve(&mut self) -> ACVMStatus<F> {
+    pub fn solve(&mut self) -> &ACVMStatus<F> {
         while self.status == ACVMStatus::InProgress {
             self.solve_opcode();
         }
-        self.status.clone()
+        &self.status
```

Comment:
What is the reasoning behind borrowing the status here? It looks like we just end up calling clone ay every call site anyway.